packages/dev/core/src/FlowGraph/flowGraphPath.ts

```diff
@@ -0,0 +1,104 @@
+import { RegisterClass } from "../Misc/typeStore";
+import type { FlowGraphContext } from "./flowGraphContext";
+
+const SEPARATORS = /\/|\./;
+/*
+ * This class represents a path of type /x/{y}/z/.../w that is evaluated
+ * on a target object. The string between curly braces ({y} in the example)
+ * is a special template string that is replaced during runtime.
+ */
+export class FlowGraphPath {
+    private _path: string;
+    private _templateSubstitutions: {
+        [key: string]: number;
+    } = {}; // this is a map of template strings to values that are substituted during runtime
+    public hasTemplateStrings: boolean = false;
+
+    constructor(path: string) {
+        this._path = path;
+        const templateStrings = this._getTemplateStringsInPath(path);
+        this.hasTemplateStrings = templateStrings.length > 0;
+        for (const templateString of templateStrings) {
+            // Part of the path enclosed in curly braces = template string, we'll set a default
+            // value on the template substitutions for now
+            this._templateSubstitutions[templateString] = NaN;
+        }
+    }
+
+    getTemplateStrings(): string[] {
+        return Object.keys(this._templateSubstitutions);
+    }
+
+    private _getTemplateStringsInPath(path: string): string[] {
+        const splitPath = path.split(SEPARATORS).filter((part) => part !== "");
+        const templateStrings: string[] = [];
+        for (const partPath of splitPath) {
+            if (partPath.startsWith("{") && partPath.endsWith("}")) {
+                templateStrings.push(partPath.slice(1, partPath.length - 1));
+            }
+        }
+        return templateStrings;
+    }
+
+    setTemplateSubstitution(template: string, value: number) {
+        this._templateSubstitutions[template] = value;
+    }
+
+    private _evaluateTemplates(): string {
+        let finalPath = this._path.slice(0); // copy the path so we don't replace on it.

```

Comment:
strings are immutable. So unless you are setting this._path to be soemthing else, every operation you will do will create a new string. So this:
```suggestion
        let finalPath = this._path;
```
is enough